src/commands/scaling/addScaleRule/addScaleRule.ts

```diff
@@ -3,75 +3,45 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { ContainerAppsAPIClient, KnownActiveRevisionsMode, Revision } from "@azure/arm-appcontainers";
-import { uiUtils } from "@microsoft/vscode-azext-azureutils";
-import { AzureWizard, IActionContext, IAzureQuickPickItem, createSubscriptionContext, nonNullProp, nonNullValue } from "@microsoft/vscode-azext-utils";
-import type { AzureSubscription } from "@microsoft/vscode-azureresources-api";
-import { ext } from "../../../extensionVariables";
-import type { ContainerAppItem, ContainerAppModel } from "../../../tree/ContainerAppItem";
-import { ScaleRuleGroupItem } from "../../../tree/scaling/ScaleRuleGroupItem";
-import { createContainerAppsClient } from "../../../utils/azureClients";
+import { KnownActiveRevisionsMode, Scale } from "@azure/arm-appcontainers";
+import { AzureWizard, IActionContext, createSubscriptionContext, nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
+import type { ScaleRuleGroupItem } from "../../../tree/scaling/ScaleRuleGroupItem";
+import { createActivityContext } from "../../../utils/activityUtils";
 import { localize } from "../../../utils/localize";
-import { pickContainerApp } from "../../../utils/pickItem/pickContainerApp";
+import { pickScaleRuleGroup } from "../../../utils/pickItem/pickScale";
 import { AddScaleRuleStep } from "./AddScaleRuleStep";
 import type { IAddScaleRuleContext } from "./IAddScaleRuleContext";
 import { ScaleRuleNameStep } from "./ScaleRuleNameStep";
-import { ScaleRuleTypeStep } from "./ScaleRuleTypeStep";
-
-const addScaleRuleTitle = localize('addScaleRuleTitle', 'Add Scale Rule');
+import { ScaleRuleTypeListStep } from "./ScaleRuleTypeListStep";
 
 export async function addScaleRule(context: IActionContext, node?: ScaleRuleGroupItem): Promise<void> {
-    const { subscription, containerApp, revision } = node ?? await getContainerAppAndRevision(context);
+    const item: ScaleRuleGroupItem = node ?? await pickScaleRuleGroup(context, { autoSelectDraft: true });
+    const { subscription, containerApp, revision } = item;
+
+    // Branching path reasoning: https://github.com/microsoft/vscode-azurecontainerapps/blob/main/src/commands/revisionDraft/README.md
+    let scale: Scale | undefined;
```

Comment:
We're going to have to duplicate this logic a lot (any time we add/update the draft). I think we should go ahead and extract this out as a util function.

The function should just return `revision` or `template` depending on the revision type.